flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/plan/QueryOperationConverter.java

```diff
@@ -130,8 +138,39 @@ public RelNode visit(QueryOperation other) {
 		}
 
 		@Override
-		public <U> RelNode visit(TableSourceQueryOperation<U> tableSourceTable) {
-			throw new UnsupportedOperationException("Unsupported now");
+		public <U> RelNode visit(TableSourceQueryOperation<U> tableSourceOperation) {
+			TableSource<?> tableSource = tableSourceOperation.getTableSource();
+			boolean isBatch = tableSourceOperation.isBatch();
```

Comment:
This will never be true in Blink. You should use the flag on `TableSource`. The `isBatch` tells if the `TableSource` is `StreamTableSource` or `BatchTableSource`.

Reply:
in blink, we always use `StreamTableSource` and use the `isBounded` flag ? do i understand correctly ?

Reply:
yes

Reply:
There is a special case: `LookupableTableSource`, which extends `TableSource`. so the `isBatch` will also be used here if a `TableSourceQueryOperation` wraps a `LookupableTableSource`.